src/XFile.cpp

```diff
@@ -75,22 +75,22 @@ std::string XFile::AppendToFilename(const std::string& filename, const std::stri
 	return newPath.string();
 }
 
-std::vector<std::string> XFile::GetFilesFromDirectory(const std::string& directory)
+std::vector<std::string> XFile::GetFilenamesFromDirectory(const std::string& directory)
 {
 	// Brett208 6Aug17: Creating a path with an empty string will prevent the directory_iterator from finding files in the current relative path.
 	auto pathStr = directory.length() > 0 ? directory : "./";
 
 	std::vector<std::string> filenames;
 	for (const auto& entry : fs::directory_iterator(pathStr)) {
-		filenames.push_back(entry.path().string());
+		filenames.push_back(GetFilename(entry.path().generic_string()));
```

Comment:
Ahh yes, this would be the simple way to do it for the non-recursive case. For some reason I was expecting this line to be a little more complicated. In the past, I've done this by cropping the first `n` characters from the path, where `n` is the number of characters in the search base folder. That would work for the recursive case. It does however have the assumption that all paths returned are prefixed with the search base folder, which is generally true.